### ISSUE TITLE
Support CSS conventions/Import option for Custom-Attributes

### ISSUE THREAD

**Issue by HamedFathi** [aurelia/aurelia#991]
# 🙋 Feature Request

As I recall correctly, There is no option to add CSS files to Custom-Attribute based on conventions or any option like what we have in custom elements.

![image](https://user-images.githubusercontent.com/8418700/97478343-09ff0a80-1966-11eb-831e-f7a6527e948d.png)

Sometimes we design attributes that are based entirely on CSS files.

## 🤔 Expected Behavior

Use CSS files in Custom-Attribute as easy as what we have in custom elements. (There is no HTML file here to import CSS files)

## 😯 Current Behavior

There is no option available.

**Comment by fkleuver** [comment]
We need to do a bit more with custom attributes and css in the context of ShadowDOM anyway. If placing a custom attribute on a ShadowDOM element automatically adds that css to the element's `adoptedStyleSheets` then that would make this an especially neat feature that would further drive ShadowDOM adoption.

Conventions alone can't do that though. We'll need to add a customAttribute API for associating css at runtime, so the runtime can take care of putting its css into its target element.

I'd say let's do it :)

**Comment by bigopon** [comment]
Supporting this one should mean we also support app wide CSS, as `app.css` is stylesheet specific to the app component. What about we apply convention specific for our entry, like `index.css`?

```
📄 index.ts
📄 index.css
📁 attributes
    📄 my-attribute.ts
    📄 my-attribute.css
```
cc @3cp thoughts?

Compiling entry related CSS might be an issue though, because we will need to modify registration code to support it, which could be wild, maybe we can let application do the registration themselves.

**Comment by 3cp** [comment]
If index.ts is a custom element, conventions will do the work.

If index.ts is not a CE, conventions plugin shouldn't touch it. It sounds like a neat feature, but in reality you don't want to create a feature that surprises users.

Also you are assuming index.ts is the starting point of the app. It is quite hard for static analysis to find out entry module as different bundlers (including future bundlers) have different way to setup.

**Comment by bigopon** [comment]
> Also you are assuming index.ts is the starting point of the app. It is quite hard for static analysis to find out entry module as different bundlers (including future bundlers) have different way to setup.

yeah, hence I mentioned it's wild, especially when we try to leave things alone as much as possible, at least for now. Maybe we can just do it for attributes only:

```ts
import { useCss } from '@aurelia/runtime-html';
import css from './my-attribute.css';

@customAttribute({
  ...
})
export class MyCustomAttribute {

}

```
or
```ts
@useCss(css)
export class MyCustomAttribute {

}
```
